Gems/SimulationInterfaces/Code/Source/Clients/ROS2SimulationInterfacesSystemComponent.cpp

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "ROS2SimulationInterfacesSystemComponent.h"
+
+#include <Actions/SimulateStepsActionServerHandler.h>
+#include <AzCore/std/string/string.h>
+#include <Services/ROS2ServiceBase.h>
+#include <SimulationInterfaces/ROS2SimulationInterfacesRequestBus.h>
+
+#include <ROS2/ROS2Bus.h>
+#include <SimulationInterfaces/ROS2SimulationInterfacesTypeIds.h>
+
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzCore/std/smart_ptr/make_shared.h>
+
+namespace ROS2SimulationInterfaces
+{
+
+    namespace
+    {
+        template<typename T>
+        void RegisterInterface(
+            AZStd::unordered_map<AZStd::string, AZStd::shared_ptr<IROS2HandlerBase>>& interfacesMap, rclcpp::Node::SharedPtr ros2Node)
+        {
+            AZStd::shared_ptr handler = AZStd::make_shared<T>();
+            handler->Initialize(ros2Node);
```

Comment:
Is there a reason the handler doesn't just accept ros2Node in constructor?

Reply:
To make implementation of services easier (no need to call base constructor).